### PR TITLE
Fix testautomation_pixels

### PR DESCRIPTION
**Pull request by 1bsyl**
we can create all pixel format, including SDL_PIXELFORMAT_UNKNOWN (as before). 
Check for really unknown pixel format. (eg != SDL_PIXELFORMAT_UNKNOWN)

### CODE REVIEW

File: src/video/SDL_pixels.c [hunk 1]
```diff
@@ -516,6 +572,12 @@ SDL_CreatePixelFormat(Uint32 pixel_format)
 {
     SDL_PixelFormat *format;
 
+    /* Only allow a known format, or explicitly SDL_PIXELFORMAT_UNKNOWN */
+    if (! SDL_IsPixelFormatValid(pixel_format)) {
```
Comment (slouken) [inline src/video/SDL_pixels.c:576]: Please remove the space between '!' and SDL

Comment (1bsyl) [reply src/video/SDL_pixels.c]: done!

File: src/video/SDL_pixels.c [hunk 2]
```diff
@@ -134,6 +134,62 @@ SDL_GetPixelFormatName(Uint32 format)
...
+        CASE(SDL_PIXELFORMAT_YV12)
+        CASE(SDL_PIXELFORMAT_IYUV)
+        CASE(SDL_PIXELFORMAT_YUY2)
+        CASE(SDL_PIXELFORMAT_UYVY)
+        CASE(SDL_PIXELFORMAT_YVYU)
+        CASE(SDL_PIXELFORMAT_NV12)
+        CASE(SDL_PIXELFORMAT_NV21)
+        CASE(SDL_PIXELFORMAT_EXTERNAL_OES)
+
+        /* also UNKNOW */
```
Comment (slouken) [inline src/video/SDL_pixels.c:184]: `UNKNOWN`

File: src/video/SDL_pixels.c [hunk 3]
```diff
@@ -516,6 +572,12 @@ SDL_CreatePixelFormat(Uint32 pixel_format)
 {
     SDL_PixelFormat *format;
 
+    /* Only allow a known format, or explicitly SDL_PIXELFORMAT_UNKNOWN */
+    if (!SDL_IsPixelFormatValid(pixel_format)) {
```
Comment (slouken) [inline src/video/SDL_pixels.c:576]: No, this is explicitly designed to support unknown formats that use the pixelformat layout. See SDL_GetMasksForPixelFormatEnum() for details.

File: test/testautomation_pixels.c [hunk 4]
```diff
@@ -158,7 +154,7 @@ int pixels_allocFreeFormat(void *arg)
             SDLTest_AssertCheck(result->format == format, "Verify value of result.format; expected: %" SDL_PRIu32 ", got %" SDL_PRIu32, format, result->format);
             SDLTest_AssertCheck(result->BitsPerPixel > 0, "Verify value of result.BitsPerPixel; expected: >0, got %u", result->BitsPerPixel);
             SDLTest_AssertCheck(result->BytesPerPixel > 0, "Verify value of result.BytesPerPixel; expected: >0, got %u", result->BytesPerPixel);
-            if (result->palette != NULL) {
+            if (result->palette != NULL && ! SDL_ISPIXELFORMAT_FOURCC(result->format)) {
```
Comment (slouken) [inline test/testautomation_pixels.c:157]: Please remove the space between '!' and SDL

File: src/video/SDL_pixels.c (hunk 1 above)
Comment (1bsyl) [reply src/video/SDL_pixels.c]: done

File: src/video/SDL_pixels.c (hunk 2 above)
Comment (1bsyl) [reply src/video/SDL_pixels.c]: reverted

File: test/testautomation_pixels.c (hunk 4 above)
Comment (1bsyl) [reply test/testautomation_pixels.c]: done

**Pull request merged**